dataland-frontend/tests/e2e/specs/community/SingleDataRequest.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  premium_user_name,
+  premium_user_pw,
+  reader_name,
+  reader_pw,
+  uploader_name,
+  uploader_pw,
+} from "@e2e/utils/Cypress";
+import { type Interception } from "cypress/types/net-stubbing";
+import { type SingleDataRequest } from "@clients/communitymanager";
+import { describeIf } from "@e2e/support/TestUtility";
+import { DataTypeEnum, type StoredCompany } from "@clients/backend";
+import { getKeycloakToken } from "@e2e/utils/Auth";
+import { generateDummyCompanyInformation, uploadCompanyViaApi } from "@e2e/utils/CompanyUpload";
+
+describeIf(
+  "As a premium user, I want to be able to navigate to the single data request page and submit a request",
+  {
+    executionEnvironments: ["developmentLocal", "ci", "developmentCd"],
+  },
+  () => {
+    const uniqueCompanyMarker = Date.now().toString();
+    const testCompanyName = "Company-for-single-data-request" + uniqueCompanyMarker;
+    let testStoredCompany: StoredCompany;
+    beforeEach(() => {
+      cy.ensureLoggedIn(uploader_name, uploader_pw);
+      getKeycloakToken(uploader_name, uploader_pw).then((token: string) => {
+        return uploadCompanyViaApi(token, generateDummyCompanyInformation(testCompanyName)).then((storedCompany) => {
+          testStoredCompany = storedCompany;
+        });
+      });
+      cy.ensureLoggedIn(premium_user_name, premium_user_pw);
+    });
+
+    it("Navigate to the single request page via the company cockpit", () => {
+      cy.visitAndCheckAppMount(`/companies/${testStoredCompany.companyId}`);
+      cy.get('[data-test="singleDataRequestButton"]').should("exist").click();
+      cy.url().should("contain", `/singledatarequest/${testStoredCompany.companyId}`);
+    });
+
+    it("Fill out the request page and check correct validation, request and success message", () => {
+      cy.intercept("POST", "**/community/requests/single").as("postRequestData");
+      cy.visitAndCheckAppMount(`/singleDataRequest/${testStoredCompany.companyId}`);
+      checkCompanyInfoSheet();
+      checkValidation();
+      chooseReportingPeriod();
+      chooseFramework();
+
+      cy.get('[data-test="contactEmail"]').type("example@Email.com");
+      cy.get('[data-test="dataRequesterMessage"]').type("Some message");
+
+      cy.get("button[type='submit']").should("exist").click();
```

Comment:
This line is used 3 times in an identical shape in the test, one could therefore outsource it to an own small function in order avoid duplication (as done for function `checkCompanyInfoSheet`)

Reply:
will do

Reply:
done